homeassistant/components/weather/yweather.py

```diff
@@ -33,23 +36,31 @@
     'fog': [19, 20, 21, 22, 23],
     'hail': [17, 18, 35],
     'lightning': [37],
-    'lightning-rainy': [38, 39],
+    'lightning-rainy': [38, 39, 47],
     'partlycloudy': [44],
     'pouring': [40, 45],
     'rainy': [9, 11, 12],
     'snowy': [8, 13, 14, 15, 16, 41, 42, 43],
-    'snowy-rainy': [5, 6, 7, 10, 46, 47],
-    'sunny': [32],
+    'snowy-rainy': [5, 6, 7, 10, 46],
+    'sunny': [32, 33, 34],
     'windy': [24],
     'windy-variant': [],
     'exceptional': [0, 1, 2, 3, 4, 25, 36],
 }
 
+_CONDITION_CLASSES_MAX = 50
+
+CONDITION_CLASSES_LIST = [str(x) for x in range(0, _CONDITION_CLASSES_MAX)]
+
+for cond, condlst in CONDITION_CLASSES.items():
+    for condi in condlst:
+        CONDITION_CLASSES_LIST[condi] = cond
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Optional(CONF_WOEID, default=None): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    vol.Optional(CONF_FORECAST, default=0):
-        vol.All(vol.Coerce(int), vol.Range(min=0, max=5)),
+    vol.Optional(CONF_FORECAST, default=5):
```

Comment:
The forecast in yaml works different from the old one, so I think maybe its better to change `forecast` to something like `forecast_days`.